src/rdkafka_cgrp.c

```diff
@@ -823,7 +829,7 @@ rd_kafka_cgrp_assignor_handle_Metadata_op (rd_kafka_t *rk,
         if (rkcg->rkcg_join_state != RD_KAFKA_CGRP_JOIN_STATE_WAIT_METADATA)
                 return RD_KAFKA_OP_RES_HANDLED; /* From outdated state */
 
-        if (!rkcg->rkcg_group_leader.protocol) {
+        if (!rkcg->rkcg_group_leader.members) {
```

Comment:
Looks like it should perhaps check rkcg_assignor instead?

Reply:
this will also be set in the normal members (which are what requires the state), so no. i will add a comment to `rkcg_assignor` in `rdkafka_cgrp.h` to note this.